packages/flutter_tools/lib/src/asset.dart

```diff
@@ -203,6 +231,27 @@ class _Asset {
 
   @override
   String toString() => 'asset: $assetEntry';
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(other, this))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final _Asset otherAsset = other;
+    return otherAsset.base == base
+        && otherAsset.assetEntry == assetEntry
+        && otherAsset.relativePath == relativePath
+        && otherAsset.source == source;
+  }
+
+  @override
+  int get hashCode {
+    return base.hashCode
+        ^assetEntry.hashCode
```

Comment:
nitty nit nit: we really prefer it when the code is vertically aligned cleanly, as in:
```dart
    return base.hashCode
￼         ^ assetEntry.hashCode
         ^ relativePath.hashCode
         ^ source.hashCode;
￼```